### PR TITLE
#190/show required fields lables on add item page by default

### LINKED ISSUE #190: Show required fields lables on add item page by default
# Expected Behaviour

```
Required fields lables are shown on add item page by default
```

# Current Behaviour

```
Required fields lables are shown on add item page only after picking an image to upload
```

## Steps to Reproduce
```
1. Run the application locally 
2. Log in as a user 
3. Go to add item page http://localhost:3000/add-item 
4. Pick an image to upload (a valid jpeg file) 
5. Click  "add your item" button 
```
Screenshot 2024-04-29 at 00 29 00

### PR DESCRIPTION
**Pull request by AlexVOiceover**
# Description

Closes #190
The add item form now shows correctly the fields that need to be filled even when an image wasn't attached yet. The previous behaviour was forcing the user to have an image uploaded.

### Files changed

app/(dashboard)/add-item/page.tsx
Makes the comprobation of the image being uploaded

components/form/UploadImageInput.tsx
Will pass as a prop a flag to specify if the image was uploaded (onImageUpload)

app/(dashboard)/add-item/pageTypes.ts
This file was created to extend the type PartialItem to include the boolean 'imageUploaded'

### UI changes

image

# Tests

 Manual tested. Try to create item without image and without some required fields

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,4 +1,9 @@
 {
+  "settings": {
```
Comment (eliazzo) [inline .eslintrc.json:2]: Is this file change intentional? If so, what is this doing

File: app/(dashboard)/add-item/page.tsx [hunk 2]
```diff
@@ -62,6 +72,12 @@ const AddItemPage = () => {
   const category = watch('item_type');
   const isWillingToPostChecked = watch('postable');
   const isPickUpChecked = watch('collectible');
+  const isPostageCovered = watch('postage_covered');
+  const isImageUploaded = watch('imageUploaded');
+
+  const handleImageUpload = (isUploaded: boolean) => {
+    setValue('imageUploaded', isUploaded); // Update and validate the field
```
Comment (eliazzo) [inline app/(dashboard)/add-item/page.tsx:79]: remove this comment :)

File: components/form/UploadImageInput.tsx [hunk 3]
```diff
@@ -78,17 +78,18 @@ const UploadImageInput: React.FC<UploadImageProps> = ({
...
+        id='image'
         className='pl-14'
         type='file'
         name='image'
         onChange={(e) => imageFileUpload(e)}
-        required={isRequired}
+        // required={isRequired}
       />
-      {!isImageUploaded && (
+      {/*!isImageUploaded && (
```
Comment (eliazzo) [inline components/form/UploadImageInput.tsx:88]: remove comments

File: components/form/UploadImageInput.tsx [hunk 4]
```diff
@@ -58,7 +55,8 @@ const UploadImageInput: React.FC<UploadImageProps> = ({
         const imagePath = CDN + userId + '/' + imageName;
 
         setImageSrc(imagePath);
-        setIsImageUploaded(true);
+        //setIsImageUploaded(true);
```
Comment (eliazzo) [inline components/form/UploadImageInput.tsx:58]: remove comments/ unused code

File: app/(dashboard)/add-item/page.tsx [hunk 5]
```diff
@@ -220,20 +246,22 @@ const AddItemPage = () => {
...
-          isRequired={true}
         />
         {generalError && <p className='error-message'>{generalError}</p>}
-
+        {!isImageUploaded && <p className='error-message'>Image is required</p>}
         <ButtonRounded type='submit'>ADD YOUR ITEM</ButtonRounded>
       </form>
     </div>
   );
+  console.log(isPostageCovered, setValue, isImageUploaded);
```
Comment (camelPhonso) [inline app/(dashboard)/add-item/page.tsx:264]: 🧹 console logs need to be removed

Comment (eliazzo) [inline app/(dashboard)/add-item/page.tsx:264]: remove logs

File: components/form/UploadImageInput.tsx [hunk 6]
```diff
@@ -7,7 +7,7 @@ import { v4 as uuidv4 } from 'uuid';
 type UploadImageProps = {
   setImageSrc: (src: string) => void;
   setError?: (error: string) => void;
-  isRequired?: boolean;
+  onImageUpload: (isUploaded: boolean) => void;
```
Comment (eliazzo) [inline components/form/UploadImageInput.tsx:10]: should this just be the setter for a new state?

File: components/form/UploadImageInput.tsx [hunk 7]
```diff
@@ -21,11 +21,11 @@ const CDN =
 const UploadImageInput: React.FC<UploadImageProps> = ({
   setImageSrc,
   setError,
-  isRequired: isRequired = false,
+  onImageUpload,
 }) => {
   const supabase = createClientComponentClient();
   const [userId, setUserId] = useState('');
-  const [isImageUploaded, setIsImageUploaded] = useState(false);
+  // const [isImageUploaded, setIsImageUploaded] = useState(false);
```
Comment (eliazzo) [inline components/form/UploadImageInput.tsx:28]: remove comments/ unused code

File: app/(dashboard)/add-item/pageTypes.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+import { PartialItem } from '@/types/supabaseTypes';
+
+//This type extends PartialItem with the boolean imaheUploaded
```
[the remark this answers is not in the document]
Comment (camelPhonso) [reply app/(dashboard)/add-item/pageTypes.ts:3]: That was supposed to show the ghostbusters - to tell you to delete comments 👻

File: app/(dashboard)/add-item/page.tsx (hunk 2 above)
Comment (eliazzo) [inline app/(dashboard)/add-item/page.tsx:79]:
```suggestion
    setValue('imageUploaded', isUploaded); 
```
remove comment

File: .eslintrc.json [hunk 9]
```diff
@@ -1,4 +1,9 @@
...
+    "react": {
+      "version": "detect"
+    }
+  },
```
Comment (camelPhonso) [inline .eslintrc.json:6]:
```suggestion

```

File: .gitignore [hunk 10]
```diff
@@ -18,7 +18,7 @@ playwright/.auth/user.json
 /build
 
 # misc
-.DS_Store
+.DS_StoreThe
```
Comment (camelPhonso) [inline .gitignore:21]:
```suggestion
.DS_Store
```

**Review by nichgalzin — APPROVED** [review]

**Comment by nichgalzin** [comment]
Hey @AlexVOiceover, 
    Can you please resolve the issues on on PR with the example .env file. Please and thanks. Then we can merge this PR.

**Comment by AlexVOiceover** [comment]
Hi @nichgalzin and team,

I have resolved the conflicts by merging the latest changes from the `dev` branch into my feature branch and deleting the `.env.example` file as per the upstream changes. The PR should now be ready for merging.

Thank you!

File: playwright-report/index.html
Comment (camelPhonso) [inline playwright-report/index.html:1]: We no longer use Playwright on the code base so I presume this is an old report that got staged. Can you remove this from the PR please?

**Review by camelPhonso — CHANGES_REQUESTED** [review]
Thanks for the update Alex.

Asked to remove an unnecessary file here.

**Comment by AlexVOiceover** [comment]
The comprobation of the mandatory fields on the form and the presence of the image cannot be done in separate react components, that way it will only perform the "outer" component once that the nested one was resolved.
All comprobations were moved to the parent (AddNewItemForm.tsx).
For a similar reason it was needed to add the function handleFormSubmit as some kind of middleware. It will update isSubmitAttempted before trying to submit, so all the form mandatory fields comprobations and the error messages for the checkboxes and the image will be done at once.
Without this function,  submitHandler will check the mandatory fields, and only after everything is right will carry on with checking the image.

setError and isRequired were left on the props for compatibility with other potential components that use this functionality, these should be udpated.

**Comment by camelPhonso** [comment]
Closing this PR as the issue has been handed over to another contributor

**Pull request closed without merging**